mesh/pncmesh.cpp

```diff
@@ -1088,8 +1100,14 @@ void ParNCMesh::GetFaceNeighbors(ParMesh &pmesh)
 
             bool sloc = (sfe.rank == MyRank);
             bool mloc = (mfe.rank == MyRank);
-            if (sloc == mloc) { continue; }
+            if (sloc == mloc // both or neither face is owned by this processor
+                || sf.index < 0) // the face is degenerate (i.e. a face-edge constraint)
+            {
+               continue;
+            }
```

Comment:
Sorry, I have to ask, but any idea why this should work for hexahedra, but not for tetrahedra? Could you try for higher order space if this still gives the correct results?

Reply:
Re higher order: rerunning the original failing case you reported,
```
mpirun -np 6 ./ex15p -m ../data/escher-p2.mesh -r 1 -est 1 -e 0.00001
```
did not fail for me after these fixes. Additionally I had identified that the same bug had been triggered for P1 meshes previously, so appeared unrelated to the order of the mesh, but the topology.

Reply:
I see. The solution looks still wrong on this mesh in this branch. I think it might be interesting to see if the error can also be triggered with wedges then (because it should, if it is purely related to the faces). I could not find time to check the logic for wedges yet. However, I am still somewhat suspecting that the serial implementation might contain something spurious. I always had trouble with the Kelly error estimator (basically the jump in the fluxes over the all faces of an element) with tets, even in serial, but I could not find time to look into the details there.